qiskit/transpiler/passes/routing/sabre_swap.py

```diff
@@ -203,10 +222,8 @@ def run(self, dag):
             for node in front_layer:
                 if len(node.qargs) == 2:
                     v0, v1 = node.qargs
-                    # Accessing layout._v2p directly to avoid overhead from __getitem__ and a
-                    # single access isn't feasible because the layout is updated on each iteration
                     if self.coupling_map.graph.has_edge(
-                        current_layout._v2p[v0], current_layout._v2p[v1]
+                        layout.get_item_logic(v0._index), layout.get_item_logic(v1._index)
```

Comment:
`Bit._index` is deprecated (or more, it's accessing the protected member that backs deprecated behaviour).

Reply:
Yeah I did that to avoid the deprecation warning :P It's mostly because we're working with all flat registers directly and I need to work with the raw index on the rust side this was just the easiest and fastest method. The only other way to do this is to do `canonical_register.index(v0)` or `_bit_indices(v0)` which would work but added a dict lookup while the `_index` attribute is slotted. But I'll change this in the interest of future compatibility.

Reply:
Fixed in: https://github.com/Qiskit/qiskit-terra/pull/7977/commits/eef49df7da8951cbc52364ab58d44ab4ad6ffd62